scio-core/src/main/scala/com/spotify/scio/experimental/package.scala

```diff
@@ -85,8 +86,18 @@ package object experimental {
         // newSource can be either table or query
         val table = scala.util.Try(BigQueryIO.parseTableSpec(newSource)).toOption
         if (table.isDefined) {
+          val newTableSchema = bigQueryClient.getTableSchema(table.get)
+          require(areSchemasCompatible(bqt.schema, newTableSchema),
+            s"""|New source $newSource has incompatible schema.
+                |New source schema ${newTableSchema.toString},
+                |Original schema ${bqt.schema.toString}!""".stripMargin)
           self.bigQueryTable(table.get)
         } else {
+          val newQuerySchema = bigQueryClient.getQuerySchema(newSource)
```

Comment:
This `newQuerySchema` may still be from view if query is legacy syntax. How do you plan to fix this?